processing/src/main/java/io/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -77,12 +77,13 @@ public void run(
     final int cardinality = params.getCardinality();
     int numProcessed = 0;
     while (numProcessed < cardinality) {
-      final int numToProcess = Math.min(params.getNumValuesPerPass(), cardinality - numProcessed);
+      int numToProcess = Math.min(params.getNumValuesPerPass(), cardinality - numProcessed);
```

Comment:
Suggest leaving as final and assigning values in both cases of the if statement.